qa/rpc-tests/wallet_listnotes.py

```diff
@@ -9,160 +9,308 @@
+        assert('jsoutindex' in unspent_cb[0])   # sprout (and overwinter) only
+        assert('outindex' not in unspent_cb[0]) # sapling only
+
+        # filter by address, should produce same result
+        assert_equal(unspent_cb,
+            self.nodes[0].z_listunspent(0, 9999, False, [sproutzaddr]))
 
-        # list unspent, filtering by address, should produce same result
-        unspent_cb_filter = self.nodes[0].z_listunspent(0, 9999, False, [sproutzaddr])
-        assert_equal(unspent_cb, unspent_cb_filter)
-        
-        # Generate a block to confirm shield coinbase tx
+        # Current height = 202 -> Sapling. Default address type remains Sprout
         self.nodes[0].generate(1)
         self.sync_all()
-        
-        # Current height = 202 -> Overwinter. Default address type remains Sprout
         assert_equal(202, self.nodes[0].getblockcount())
+        assert_equal(10-fee, Decimal(self.nodes[0].z_gettotalbalance()['private']))
 
-        # Send 1.0 (actually 0.9999) from sproutzaddr to a new zaddr
-        sproutzaddr2 = self.nodes[0].z_getnewaddress()
-        receive_amount_1 = Decimal('1.0') - Decimal('0.0001')
-        change_amount_9 = receive_amount_10 - Decimal('1.0')
-        assert_equal('sprout', self.nodes[0].z_validateaddress(sproutzaddr2)['type'])
-        recipients = [{"address": sproutzaddr2, "amount":receive_amount_1}]
-        myopid = self.nodes[0].z_sendmany(sproutzaddr, recipients)
+        received = self.nodes[0].z_listreceivedbyaddress(sproutzaddr, 0)
+        assert_equal(1, len(received))
+        assert_equal(my_memo,           received[0]['memo'])
+        assert_equal(Decimal('10')-fee, received[0]['amount'])
+        assert_equal(txid_1,            received[0]['txid'])
+        assert_equal(False,             received[0]['change'])
+        assert('jsoutindex' in received[0])   # sprout (and overwinter) only
+        assert('outindex' not in received[0]) # sapling only
+
+        # beginning with Sapling, change is allowed from a coinbase transfer
+        myopid = self.nodes[0].z_sendmany(mining_addr,
+            [{'address':saplingzaddr1, 'amount':Decimal('8')-fee}])
         txid_2 = wait_and_assert_operationid_status(self.nodes[0], myopid)
         self.sync_all()
-        
-        # list unspent, allowing 0conf txs
         unspent_tx = self.nodes[0].z_listunspent(0)
-        assert_equal(len(unspent_tx), 2)
-        # sort low-to-high by amount (order of returned entries is not guaranteed)
+        # sort low-to-high by amount (order of returned entries is unpredictable)
         unspent_tx = sorted(unspent_tx, key=lambda k: k['amount'])
-        assert_equal(False,             unspent_tx[0]['change'])
-        assert_equal(txid_2,            unspent_tx[0]['txid'])
+        assert_equal(2, len(unspent_tx))
+
+        # TODO: should default memo be no_memo? (it is for Sprout) See PR #3518
+        assert_equal(zero_memo,         unspent_tx[0]['memo'])
```

Comment:
I think this may be a bug in `z_sendmany()`; at least it works differently in Sapling (memo is all zeros) compared with Sprout ("empty memo" that starts with `0xf6`).

Reply:
@str4d Do you know the answer to this?

Reply:
Huh, I thought we had fixed this?

Reply:
Oh, we only fixed it for change.